qemu/tests/check_basepage_size.py

```diff
@@ -0,0 +1,61 @@
+import os
+
+from avocado.utils import process
+
+from virttest import data_dir
+from virttest import error_context
+
+from virttest.remote import scp_to_remote
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Check the basic page size.
+    Steps:
+    1) Check system configuration basic page size on host.
+    2) Get the actual allocation size of basic page on host.
+    3) Boot a guest on the host.
+    4) Check system configuration basic page size on guest.
+    5) Get the actual allocation size of basic page on guest.
+
+    :params test: QEMU test object.
+    :params params: Dictionary with the test parameters.
+    :params env: Dictionary with test environment.
+    """
+    get_page_size = params.get('get_page_size')
+    username = params.get('username')
+    password = params.get('password')
+    port = params.get('file_transfer_port')
+    guest_path = params.get('guest_path')
+    build_get_pagesize_cmd = params.get('build_get_pagesize_cmd')
+    get_pagesize_cmd = params.get('get_pagesize_cmd')
+
+    test.log.info("Check system configuration basic page size on host.")
+    host_page_size = process.system_output(get_page_size).decode()
+    if host_page_size != params.get("page_size"):
+        test.fail("Host page size is %s not as expected." % host_page_size)
+
+    test.log.info("Get the actual allocation size of basic page on host.")
+    host_path = os.path.join(data_dir.get_deps_dir(), 'memory')
+    host_page_size = process.system_output(
```

Comment:
In fact, PAGESIZE is just a configuration option of the kernel, but on arm64 the actual ones that take effect are '#define KEXEC_CONTROL_PAGE_SIZE 4096' and 'CONFIG_ARM64_PAGE_SHIFT'.
So I think in addition to looking at the configuration, we should also programmatically check the page size that the system actually allocates to us.

Reply:
Actually, I am not sure about it. Just found a man page: https://man7.org/linux/man-pages/man2/getpagesize.2.html

So maybe we can discuss it with other people offline.

Reply:
And if you want to check it, `printf("The basic page size = %d\n",psize);` can be replaced with `printf("%d\n",psize);` I think, so that `grep [0-9] |awk -F " " '{print $6}'` is not necessary.